openstudiocore/src/model/SiteGroundTemperatureDeep.hpp

```diff
@@ -0,0 +1,183 @@
+/**********************************************************************
+ *  Copyright (c) 2008-2016, Alliance for Sustainable Energy.
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#ifndef MODEL_SITEGROUNDTEMPERATUREDEEP_HPP
+#define MODEL_SITEGROUNDTEMPERATUREDEEP_HPP
+
+#include "ModelAPI.hpp"
+#include "ModelObject.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  class SiteGroundTemperatureDeep_Impl;
+
+} // detail
+
+/** SiteGroundTemperatureDeep is a ModelObject that wraps the OpenStudio IDD object 'OS_Site_GroundTemperature_Deep'. */
+class MODEL_API SiteGroundTemperatureDeep : public ModelObject {
```

Comment:
@macumber this is mostly a copy of the `SiteGroundTemperatureBuildingSurface` files. I plan to add two more. I'm thinking it would be beneficial to have a base class that implements all these methods that all four of the models can derive from instead of duplicating them all. I'm a little fuzzy on how to implement all that, though. Is this being done somewhere else in OpenStudio that I could copy the pattern?